### PR TITLE
README Helper / Sponsor Generation Script / Workflow

### PR DESCRIPTION
**Pull request by ElectronicsArchiver**
[Badge Suggestions]: https://img.shields.io/badge/Ｒｅｐｏｓｉｔｏｒｙ_Ｓｕｇｇｅｓｔｉｏｎｓ-609926?style=for-the-badge&logoColor=white&logo=GitHub
[Badge Changes]: https://img.shields.io/badge/Ｃｈａｎｇｅｓ-D81939?style=for-the-badge&logoColor=white&logo=GitLFS
[Badge Robot]: https://img.shields.io/badge/Ｉ＇ｍ_Ｈｕｍａｎ-c84f8c?style=for-the-badge&logoColor=white&logo=RuboCop
[Badge Ask]: https://img.shields.io/badge/Ｐｌｅａｓｅ_Ａｓｋ_Ｍｅ-73398D?style=for-the-badge&logoColor=white&logo=AskUbuntu

[License Badges]: https://github.com/MarkedDown/Licenses
[Widgets]: https://github.com/MarkedDown/Disable/blob/main/Features/Widgets.md
[Tabs]: https://github.com/MarkedDown/Disable/blob/main/Features/Tabs.md
[Wiki]: https://github.com/MarkedDown/Wiki

<br>
<br>

<div align = center>

[<kbd> <br> Please Check The **Preview** <br> </kbd>][Preview]

</div>
  
<br>
<br>

## ![Badge Changes]

- Extracted sponsors / helpers from README
  
  ⤷ `docs/Helpers.yaml`, `docs/Sponsors.yaml`

- Wrote deno tool to generate `README.md`
  from `docs/TEMPLATE.md`.

  - Refactored
  - Tested

- Wrote workflow to automate generation

  - Might want to change trigger from `push` to `pull_request`
  - Might want to adjust commit message / author / ..

- Documented Tool

<br>
<br>

## ![Badge Ask]

- ***If you have any questions.***

- ***In case you do not agree with some of the changes made in this PR***, <br>
  *please first let me know about them, before disregarding it entirely so*<br>
  *we can possibly find a middle ground and have it not go to waste.*

<br>
<br>

## ![Badge Robot]

<kbd> <br>            No, I'm not a robot.            <br> </kbd>

<kbd> <br>    Yes, you are not the first.    <br> </kbd>

<kbd> <br>     I just made a PR template.     <br> </kbd>

<br>
<br>

[Preview]: https://github.com/ElectronicsArchiver/react-hook-form

### CODE REVIEW

**Comment by bluebill1049** [comment]
This is quite awesome! Loving it, could you prettier all the code that you introduced, so the code style can be consistent, feel free to update the prettier config as well. 👯

File: scripts/README/Insert.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (bluebill1049) [inline scripts/README/Insert.js:1]: could we include those files for code prettier as well? thanks.

Comment (ElectronicsArchiver) [reply scripts/README/Insert.js:1]: I'm actually not sure why prettier doesn't trigger for `.js` files..

Comment (bluebill1049) [reply scripts/README/Insert.js:1]: I think just need to update this to include JS: https://github.com/react-hook-form/react-hook-form/blob/master/package.json#L33

Comment (ElectronicsArchiver) [reply scripts/README/Insert.js:1]: Oh, wow, I looked through every config but ignore the package.json x3

**Comment by ElectronicsArchiver** [comment]
As you suggested, I updated the prettier style a bit.

*Can't say I like prettier in any case~*

**Comment by bluebill1049** [comment]
> As you suggested, I updated the prettier style a bit.
> 
> _Can't say I like prettier in any case~_

could we keep the same style? without 300 plus files changes 🙏

**Comment by ElectronicsArchiver** [comment]
What? A formatting somehow changed how a code
scanning tool reacts to syntax / logic?

> const emailRegex =
      /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;

> const emailRegex =
            /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;

**Comment by ElectronicsArchiver** [comment]
Woops, wrong button~

File: scripts/README/Imports.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+        "YAML" : "https://deno.land/std@0.148.0/encoding/yaml.ts" ,
+        "Time" : "https://deno.land/std@0.148.0/datetime/mod.ts" ,
+        "Path" : "https://deno.land/std@0.148.0/path/mod.ts" ,
+        
+        "Previews" : "./Previews/mod.ts" ,
+        "Files" : "./Files.js" ,
+        "Link" : "./Link.js" ,
+        "Data" : "./Data.js"
+    }
+}
```
Comment (bluebill1049) [inline scripts/README/Imports.json:12]:
```suggestion
}

```

**Review by bluebill1049 — APPROVED** [review]
Thank you! This is awesome.

**Comment by bluebill1049** [comment]
Screen Shot 2022-07-14 at 9 40 29 am

could you amend the action to run on PR not on master? 🙏

**Comment by bluebill1049** [comment]
Screen Shot 2022-07-14 at 9 41 33 am

**Pull request merged**